apps/freeablo/faworld/monster.cpp

```diff
@@ -0,0 +1,46 @@
+#include "monster.h"
+#include "../fasavegame/gameloader.h"
+#include "actor.h"
+#include "diabloexe/monster.h"
+#include <boost/format.hpp>
+#include <random/random.h>
+
+namespace FAWorld
+{
+    Monster::Monster(World& world, Random::Rng& rng, const DiabloExe::Monster& monsterStats) : Actor(world), mMonsterStats(monsterStats)
+    {
+        boost::format fmt(monsterStats.cl2Path);
+        mAnimation.setAnimation(AnimState::walk, FARender::Renderer::get()->loadImage((fmt % 'w').str()));
+        mAnimation.setAnimation(AnimState::idle, FARender::Renderer::get()->loadImage((fmt % 'n').str()));
+        mAnimation.setAnimation(AnimState::dead, FARender::Renderer::get()->loadImage((fmt % 'd').str()));
+        mAnimation.setAnimation(AnimState::attack, FARender::Renderer::get()->loadImage((fmt % 'a').str()));
+        mAnimation.setAnimation(AnimState::hit, FARender::Renderer::get()->loadImage((fmt % 'h').str()));
+
+        mBehaviour.reset(new BasicMonsterBehaviour(this));
+        mFaction = Faction::hell();
+        mName = monsterStats.monsterName;
+        mSoundPath = monsterStats.soundPath;
+        mStats.mHp.max = monsterStats.maxHp;
+        mStats.mHp.current = rng.randomInRange(monsterStats.minHp, monsterStats.maxHp);
+    }
+
+    Monster::Monster(World& world, FASaveGame::GameLoader& loader) : Actor(world, loader), mMonsterStats(loader)
```

Comment:
You need to add a way to tell that the actor being loaded from a save is a monster.
We do this by registering a type here https://github.com/wheybags/freeablo/blob/606d4b98ab5a28b4ad98796410034121346171a3/apps/freeablo/faworld/world.cpp#L107

You should be able to look at what we're doing for player and do the same. To test, just save and load in a dungeon.